File: app/src/test/unitTests/java/de/htwBerlin/ois/model/repositories/localRepositories/LocalMapsRepositoryTest.java
package de.htwBerlin.ois.model.repositories.localRepositories;

import androidx.lifecycle.LiveData;


import org.junit.Test;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.extension.ExtendWith;
import org.junit.runner.RunWith;
import org.mockito.Mockito;

import java.io.File;
import java.util.ArrayList;
import java.util.List;
import java.util.logging.Handler;


import static org.junit.jupiter.api.Assertions.*;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;



public class LocalMapsRepositoryTest
{
    private File mockFile = mock(File.class);
    private File[] localFileList;


    @BeforeEach
    public void setup()
    {
        localFileList = new File[5];
        File mockFileOne = Mockito.mock(File.class);
        File mockFileTwo = Mockito.mock(File.class);
        File mockFileThree = Mockito.mock(File.class);
        File mockFileFour= Mockito.mock(File.class);
        File mockFileFive= Mockito.mock(File.class);

        when(mockFileOne.getName()).thenReturn("FileOne.map");
        when(mockFileTwo.getName()).thenReturn("FileTwo.map");
        when(mockFileThree.getName()).thenReturn("FileThree.map");
        when(mockFileFour.getName()).thenReturn("FileFour.map");
        when(mockFileFive.getName()).thenReturn("FileFive.map");

        when(mockFileOne.length()).thenReturn(123l);
        when(mockFileTwo.length()).thenReturn(123l);
        when(mockFileThree.length()).thenReturn(123l);
        when(mockFileFour.length()).thenReturn(123l);
        when(mockFileFive.length()).thenReturn(123l);

        localFileList[0] = mockFileOne;
        localFileList[1] = mockFileTwo;
        localFileList[2] = mockFileThree;
        localFileList[3] = mockFileFour;
        localFileList[4] = mockFileFive;
        when(mockFile.listFiles()).thenReturn(localFileList);
    }

}

File: app/src/test/androidTest/java/de/htwBerlin/ois/model/repositories/localRepositories/LocalMapsRepositoryTest.kt
package de.htwBerlin.ois.model.repositories.localRepositories

import androidx.arch.core.executor.testing.InstantTaskExecutorRule
import androidx.lifecycle.LiveData
import androidx.test.ext.junit.runners.AndroidJUnit4
import de.htwBerlin.ois.OneTimeObserver
import io.mockk.every
import io.mockk.mockk
import junit.framework.Assert
import org.junit.Before
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith
import java.io.File

@RunWith(AndroidJUnit4::class)
class LocalMapsRepositoryTest
{
        private lateinit var mockFile : File
        private lateinit var localFileList : Array<File>

        @get:Rule
        val instantTaskExecutorRule = InstantTaskExecutorRule()


        @Before
        fun setup() {

            val mockFileOne = mockk<File>()
            every {
                mockFileOne.getName()
            } returns "FileOne.map"
            every {
                mockFileOne.length()
            } returns 123

            val mockFileTwo = mockk<File>()
            every {
                mockFileTwo.getName()
            } returns "FileTwo.map"
            every {
                mockFileTwo.length()
            } returns 123

            val mockFileThree = mockk<File>()
            every {
                mockFileThree.getName()
            } returns "FileThree.map"
            every {
                mockFileThree.length()
            } returns 123

            val mockFileFour = mockk<File>()
            every {
                mockFileFour.getName()
            } returns "FileFour.map"
            every {
                mockFileFour.length()
            } returns 123

            val mockFileFive = mockk<File>()
            every {
                mockFileFive.getName()
            } returns "FileFive.map"
            every {
                mockFileFive.length()
            } returns 123

            localFileList = arrayOf(mockFileOne, mockFileTwo, mockFileThree, mockFileFour,  mockFileFive)

            mockFile = mockk<File>()
            every{
                mockFile.listFiles()
            } returns localFileList
        }

        @Test
        fun readFromDirectoryTest()
        {
            val localMapsRepository = LocalMapsRepository.getInstance()
            localMapsRepository.getLocalFiles(mockFile).observeOnce {
                Assert.assertEquals(5, it.size)
            }
        }

        fun <T> LiveData<T>.observeOnce(onChangeHandler: (T) -> Unit) {
            val observer = OneTimeObserver(handler = onChangeHandler)
            observe(observer, observer)
        }
}